File: Java/NetBeans/SV/src/Exe_DSSV/SVSP.java
package Exe_DSSV;

public class SVSP extends SV {

    protected String noiTT;
    protected double diemTT;

    public SVSP(String hoTen, int namSinh, double dtb, String noiTT, double diemTT) {
        super(hoTen, namSinh, dtb);
        this.noiTT = noiTT;
        this.diemTT = diemTT;
    }

    public void hienThi() {
        super.hienThi();
        System.out.println("Noi thuc tap: " + noiTT);
        System.out.println("Diem thuc tap: " + diemTT);
    }

    public String loaiSV() {
        return "SP";
    }

    public boolean equals(Object sv) {
        if (sv != null && sv instanceof SVSP b) {
            return super.equals(b)
                    && this.noiTT.equals(b.noiTT)
                    && this.diemTT == b.diemTT;
        }
        return false;
    }

    public String toString() {
        return super.toString() + "\n"
                + "Noi thuc tap: " + noiTT + "\n"
                + "Diem thuc tap: " + diemTT;
    }

    public boolean duocTN() {
        return dtb >= 5.0f && diemTT >= 7.0f;
    }
}

File: C#/QLSVTHvaSP/QLSVTHvaSP/SVSP.cs
namespace QLSVTHvaSP
{
    internal class SVSP : SV
    {
        private string _noiTT;
        private float _diemTT;

        public SVSP() : base() { }

        public SVSP(string hoTen, int namSinh, float diemTB, string noiTT, float diemTT) : base(hoTen, namSinh, diemTB)
        {
            _noiTT = noiTT;
            _diemTT = diemTT;
        }

        public string NoiTT { get => _noiTT; set => _noiTT = value; }
        public float DiemTT { get => _diemTT; set => _diemTT = value; }

        public override void HienThi()
        {
            base.HienThi();
            Console.Write($", {_noiTT}, {_diemTT}");
        }

        public override bool DuocTN()
        {
            return DiemTB >= 5 && _diemTT >= 7;
        }

        public override string LoaiSV()
        {
            return "SP";
        }
    }
}